tracing-core/src/metadata.rs

```diff
@@ -121,12 +125,14 @@ static MAX_LEVEL: AtomicUsize = AtomicUsize::new(LevelFilter::OFF_USIZE);
 impl<'a> Metadata<'a> {
     /// Construct new metadata for a span or event, with a name, target, level, field
     /// names, and optional source code location.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
While we're making breaking changes for 0.2, we might want to consider combining the file, line, module, and column data into a new `Location` struct, or something, so that the constructor doesn't have a giant pile of arguments. Something like this:
```rust
#[derive(Eq, PartialEq, Clone, Debug, Default)]
pub struct Location<'a> {
    file: Option<&'a str>,
    line: Option<u32>,
    column: Option<u32>,
    module_path: Option<&'a str>,
}

impl<'a> Location<'a> {
    pub const fn new(/* ... */) -> Self {
        // ...
    }

    // accessors for various fields...
}

impl<'a> Metadata<'a> {
    pub const fn new(name: &'static str,
        target: &'a str,
        level: Level,
        location: Location<'a>,
        fields: field::FieldSet,
        kind: Kind,
    ) -> Self {
        // ...
    }
}
```

What do you think? This isn't a blocker for merging this PR, but it might be worth doing while we have the opportunity to make a breaking change here, and it would mean we don't need to allow the clippy lint, as well as making this API a little nicer to use...

Reply:
We could probably even have a `location!` macro that expands to
```rust
Location::new(Some(file!()), Some(line!()), Some(column!()), Some!(module_path!())
```
to make the type easier to construct, but this might be better off in the `tracing` crate, for stability reasons...